app/controllers/application_controller.rb

```diff
@@ -5,7 +5,12 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  before_action :better_errors_hack, if: -> { Rails.env.development? } # Hack to make better errors work with Puma
   before_action :authenticate_user!
+  after_action :verify_authorized, unless: :devise_controller?
+  after_action :verify_policy_scoped, only: :index
```

Comment:
`verify_authorized` and `verify_policy_scoped` are after hooks that check if authorization was made in controller action, and if the result was scoped down to what the user is allowed to see. We are skipping Devise controllers because the user is not yet logged in when accessing it, so no point in authorizing.